modules/gapi/src/streaming/vpl/vpl_utils.cpp

```diff
@@ -0,0 +1,497 @@
+            [&ret](int64_t value)   { ret.Type = MFX_VARIANT_TYPE_I64;  ret.Data.I64 = value;   },
+            [&ret](float_t value)   { ret.Type = MFX_VARIANT_TYPE_F32;  ret.Data.F32 = value;   },
+            [&ret](double_t value)  { ret.Type = MFX_VARIANT_TYPE_F64;  ret.Data.F64 = value;   },
+            [&ret](void* value)     { ret.Type = MFX_VARIANT_TYPE_PTR;  ret.Data.Ptr = value;   },
+            [&ret, &name] (const std::string& value) {
+                auto parsed = get_params_from_string<mfxVariant>(name + ": " + value + "\n");
+                if (parsed.empty()) {
+                    throw std::logic_error("Unsupported parameter, name: " + name + ", value: " + value);
+                }
+                ret = *parsed.begin();
+            }), cfg_val.get_value());
+    return ret;
+}
+
+std::string mfxstatus_to_string(mfxStatus err) {
+    switch(err)
+    {
+        case MFX_ERR_NONE:
+            return "MFX_ERR_NONE";
+        case MFX_ERR_UNKNOWN:
+            return "MFX_ERR_UNKNOWN";
+        case MFX_ERR_NULL_PTR:
+            return "MFX_ERR_NULL_PTR";
+        case MFX_ERR_UNSUPPORTED:
+            return "MFX_ERR_UNSUPPORTED";
+        case MFX_ERR_MEMORY_ALLOC:
+            return "MFX_ERR_MEMORY_ALLOC";
+        case MFX_ERR_NOT_ENOUGH_BUFFER:
+            return "MFX_ERR_NOT_ENOUGH_BUFFER";
+        case MFX_ERR_INVALID_HANDLE:
+            return "MFX_ERR_INVALID_HANDLE";
+        case MFX_ERR_LOCK_MEMORY:
+            return "MFX_ERR_LOCK_MEMORY";
+        case MFX_ERR_NOT_INITIALIZED:
+            return "MFX_ERR_NOT_INITIALIZED";
+        case MFX_ERR_NOT_FOUND:
+            return "MFX_ERR_NOT_FOUND";
+        case MFX_ERR_MORE_DATA:
+            return "MFX_ERR_MORE_DATA";
+        case MFX_ERR_MORE_SURFACE:
+            return "MFX_ERR_MORE_SURFACE";
+        case MFX_ERR_ABORTED:
+            return "MFX_ERR_ABORTED";
+        case MFX_ERR_DEVICE_LOST:
+            return "MFX_ERR_DEVICE_LOST";
+        case MFX_ERR_INCOMPATIBLE_VIDEO_PARAM:
+            return "MFX_ERR_INCOMPATIBLE_VIDEO_PARAM";
+        case MFX_ERR_INVALID_VIDEO_PARAM:
+            return "MFX_ERR_INVALID_VIDEO_PARAM";
+        case MFX_ERR_UNDEFINED_BEHAVIOR:
+            return "MFX_ERR_UNDEFINED_BEHAVIOR";
+        case MFX_ERR_DEVICE_FAILED:
+            return "MFX_ERR_DEVICE_FAILED";
+        case MFX_ERR_MORE_BITSTREAM:
+            return "MFX_ERR_MORE_BITSTREAM";
+        case MFX_ERR_GPU_HANG:
+            return "MFX_ERR_GPU_HANG";
+        case MFX_ERR_REALLOC_SURFACE:
+            return "MFX_ERR_REALLOC_SURFACE";
+        case MFX_ERR_RESOURCE_MAPPED:
+            return "MFX_ERR_RESOURCE_MAPPED";
+        case MFX_ERR_NOT_IMPLEMENTED:
+            return "MFX_ERR_NOT_IMPLEMENTED";
+        case MFX_WRN_DEVICE_BUSY:
+            return "MFX_WRN_DEVICE_BUSY";
+        case MFX_WRN_VIDEO_PARAM_CHANGED:
+            return "MFX_WRN_VIDEO_PARAM_CHANGED";
```

Comment:
and this. BTW, does VPL provides its own means to convert error codes to error messages?